app/models/concerns/has_many_common.rb

```diff
@@ -10,15 +10,20 @@ def name_method
     send(self.class.attribute_name)
   end
 
+  def id_str
+    self.id.to_s
+  end
+
   module ClassMethods
     # default attribute used by *_names and *_name is :name
-    # if :name doesn't exist, :id is used, so it doesn't error out if attr_name :field is not defined
-    # most likely model will have attr_name :field to overwrite attribute_name
+    # if :name doesn't exist, :id as a string is used, so it doesn't error out
+    # if attr_name :field is not defined most likely model will have attr_name
+    # :field to overwrite attribute_name
     def attribute_name
       if has_name?
         :name
       else
-        :id
+        :id_str
```

Comment:
I think this may cause issues in the meta-programmed methods below, e.g. `"find_by_#{assoc_klass(association).attribute_name}"` will no longer work